.gitignore

```diff
@@ -19,6 +19,18 @@ src/tools/log-parser/kata-log-parser
 # Microsoft-specific
 .cargo/
 vendor/
+!src/tardev-snapshotter/vendor
+!src/tardev-snapshotter/vendor/**/vendor
+!src/tardev-snapshotter/vendor/**/build
+!src/tardev-snapshotter/.cargo
+!src/agent/vendor
+!src/agent/vendor/**/vendor
+!src/agent/vendor/**/build
+!src/agent/.cargo
+!src/utarfs/vendor
+!src/utarfs/.cargo
+!src/overlay/vendor
+!src/overlay/.cargo
```

Comment:
Seems like we're ignoring everything then re-including everything. So should we just remove all these lines then?

Reply:
We are ignoring vendor crates for other submodules such as runtime-rs, dragonball. 
We are selectively including vendor directories for the modules we build in kata-containers-cc spec of Azure Linux. 
Reference: [Vendor only what's necessary - tardev-snapshotter, agent, utarfs, overlay](https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/2568/Building-RPMs-on-a-node-and-creating-vendor-tarballs?anchor=vendor-only-what%27s-necessary---tardev-snapshotter%2C-agent%2C-utarfs%2C-overlay)